File: E-Tour/src/main/java/com/etour/common/CityNames.java
package com.etour.common;

import java.util.ArrayList;
import java.util.LinkedHashSet;
import java.util.List;
import java.util.Set;

public final class CityNames {
    private static final String SEPARATORS = "\\s+[-\u2013\u2014]\\s+|\\s*,\\s*|\\s*/\\s*|\\s+to\\s+";

    private CityNames() {
    }

    public static List<String> split(String destination) {
        List<String> out = new ArrayList<>();
        if (destination == null || destination.isBlank()) {
            return out;
        }

        Set<String> seen = new LinkedHashSet<>();
        for (String part : destination.split(SEPARATORS)) {
            String city = part.trim();
            if (city.isEmpty()) {
                continue;
            }
            if (city.length() > 100) {
                city = city.substring(0, 100);
            }
            seen.add(city);
        }
        out.addAll(seen);
        return out;
    }
}

File: ETour-DotNet/Common/CityNames.cs
using System.Text.RegularExpressions;

namespace ETour.Api.Common;

public static class CityNames
{
    private static readonly Regex Separators =
        new(@"\s+[-–—]\s+|\s*,\s*|\s*/\s*|\s+to\s+",
            RegexOptions.Compiled | RegexOptions.IgnoreCase);

    public static List<string> Split(string destination)
    {
        var outList = new List<string>();
        if (string.IsNullOrWhiteSpace(destination))
        {
            return outList;
        }

        var seen = new HashSet<string>();
        foreach (var part in Separators.Split(destination))
        {
            var city = part.Trim();
            if (city.Length == 0) continue;
            if (city.Length > 100) city = city[..100];
            if (seen.Add(city)) outList.Add(city);
        }
        return outList;
    }
}
